department-of-veterans-affairs/caseflow
# HunJerBAH/APPEALS-4133: Update Search Queries

Resolves [APPEALS-4133](https://vajira.max.gov/browse/APPEALS-4133)

### Description
As a developer, in order to satisfy the requirements for Split Appeals, I need for the system to have the ability to have more than one appeal/lane with the same Docket Number in order to implement the Split Appeals functionality.  For this to happen the complete Caseflow system needs to be updated in order to allow queries to produce, as a result, more than one appeal. 

### Acceptance Criteria
- [ ] All queries and internal functions that handle and/or are impacted by having more than one appeal with the same docket number are identified
- [ ] All queries and internal functions that handle and/or are impacted by having more than one appeal with the same docket number are corrected to consider split appeals.  
- [ ] Update functions are validated to ensure that only the correct appeal lane is updated (in the case of more than one appeal with the same docket number)

### Testing Plan
1. Go to Demo and sign in as a SSC member.
2. Go to **queue** and **search cases**. Enter a veteran file number from the database. I used 673020132
![image](https://user-images.githubusercontent.com/99915461/200672730-e5f676fa-5a6e-4c1d-bc7a-69f53bc643ca.png)
3. Click on the docket number to go to the Case Details page. 
![image](https://user-images.githubusercontent.com/99915461/200672914-191cc820-c3e4-46b2-af74-35078145916f.png)
4. Click on **Correct Issues** to navigate to the add issues page.
![image](https://user-images.githubusercontent.com/99915461/200672969-f1e45726-d991-49b2-9409-0d6f5078467a.png)
5. Click add issues. Fill out the form. Do this twice to add 2 issues to the appeal. Click **save**. The issues will appear on the appeal.
![image](https://user-images.githubusercontent.com/99915461/200673268-aa13b8be-59c7-471e-885e-6586ecaace54.png)
![image](https://user-images.githubusercontent.com/99915461/200673333-56b1c3df-ca46-4196-80c2-6149401cfa3f.png)
![image](https://user-images.githubusercontent.com/99915461/200673419-9f69e79b-4ea7-47ba-890a-5ce21071a9c1.png)
6. Click **correct issues** again to navigate to the add issues page. 
7. Click on the **split appeal** button.
8. Fill out the dropdown and select at least 1 of the issues to split off. Click **continue**.
![image](https://user-images.githubusercontent.com/99915461/200673668-e2823312-701a-4fa1-abe2-d06029bfd884.png)
9. Click **split appeal** on the next page to split the appeal.
![image](https://user-images.githubusercontent.com/99915461/200673747-1b99c9e7-ad22-4619-99e0-c962c957561f.png)
10. You should see a green success banner and only 1 issue on the appeal.
![image](https://user-images.githubusercontent.com/99915461/200673995-08875f98-778a-4f7b-8e1e-bd63fc5e3616.png)
11. Copy the veteran file number. Click **search cases** at the top. Put the veteran file number in the search bar. You should see 2 appeals with 1 issue each. Open them both up to make sure the correct issue was split. 
![image](https://user-images.githubusercontent.com/99915461/200674184-df48869c-5e82-466b-bc7d-5448604d3dda.png)
![image](https://user-images.githubusercontent.com/99915461/200674315-0cb30558-11d2-46ae-b678-a5020b12961b.png)
![image](https://user-images.githubusercontent.com/99915461/200674342-b7b153b9-8d40-45d7-b3b1-87fe646cc3e7.png)
